demo/portal/src/app/config.js

```diff
@@ -40,23 +48,40 @@ function AppConfig($stateProvider, $urlRouterProvider) {
         };
 
         $scope.navigateTo = function(state, params) {
+
           if (state) {
-            if ($mdSidenav('left').isOpen()) {
-              $mdSidenav('left').close();
+            if (params && params.authorised) {
+              if ($mdSidenav('left').isOpen()) {
+                $mdSidenav('left').close();
+              }
+              $state.go(state, params);
+            } else {
+              var message = "Unauthorised";
+              $state.go('app.unauthorised', message);
             }
-            $state.go(state, params);
+
           }
         };
 
-        $scope.hasResourceRole = function(role) {
-          return userService.hasResourceRole(role);
+        $scope.hasResourceRole = function(role, resource) {
+          var authorised = userService.hasResourceRole(role);
```

Comment:
I do not see value from that. 
For both keycloak and passport now you can switch from admin,user roles (groups) to workorder/workflow roles. 
IMHO This change may be confusing.

Reply:
it allows us to fine-grained access controll to different resources - e.g. a user can access workorders but only admin can access workorders and workflows. At the moment in Angular side we dont have interface to check if user can access a given resource, its either whole portal or nothing.

Reply:
Yeah, get your point, but that's feature of keycloak and passport. You should pass `workorders` to resource role instead of admin. 

>At the moment in Angular side we dont have interface to check if user can access a given resource, its either whole portal or nothing.

That will be planned in angular4 rewrite. We going to move from group based auth to role - workorder edit etc.